fahrschein/src/test/java/org/zalando/fahrschein/NakadiReaderTest.java

```diff
@@ -606,4 +608,29 @@ public void shouldCloseOnRuntimeException() throws IOException {
         }
     }
 
+    @Test
+    public void shouldCloseOnInactiveListener() throws IOException {
+        final ClientHttpResponse response = mock(ClientHttpResponse.class);
+        final ByteArrayInputStream inputStream = new ByteArrayInputStream("{\"cursor\":{\"partition\":\"123\",\"offset\":\"456\"},\"foo\":\"bar\",\"events\":[{\"id\":\"789\"}],\"metadata\":{\"foo\":\"bar\"}}".getBytes("utf-8"));
+        when(response.getBody()).thenReturn(inputStream);
+
+        final ClientHttpRequest request = mock(ClientHttpRequest.class);
+        when(request.execute()).thenReturn(response);
+        final HttpHeaders headers = new HttpHeaders();
+        when(request.getHeaders()).thenReturn(headers);
+
+        when(clientHttpRequestFactory.createRequest(uri, HttpMethod.GET)).thenReturn(request);
+
+        final NoBackoffStrategy backoffStrategy = new NoBackoffStrategy();
+
+        ReaderManager readerManager = (Optional<Subscription> subscription) -> false;
```

Comment:
Should the connection even be opened in this case?

Another testcase would be 

    ReaderManager readerManager = mock(ReaderManager.class);
    when(readerManager.continueReading()).thenReturn(true).thenReturn(false);

And then verify that the listener was called once.